src/tagmanager/tm_source_file.c

```diff
@@ -303,123 +305,128 @@ static gboolean init_tag_from_file_alt(TMTag *tag, TMSourceFile *file, FILE *fp)
 	return TRUE;
 }
 
-/*
- CTags tag file format (http://ctags.sourceforge.net/FORMAT)
-*/
-static gboolean init_tag_from_file_ctags(TMTag *tag, TMSourceFile *file, FILE *fp, TMParserType lang)
+static void read_ctags_file(const gchar *tags_file, TMParserType lang, GPtrArray *file_tags)
 {
-	gchar buf[BUFSIZ];
-	gchar *p, *tab;
+	tagEntry entry;
+	tagFile *f = tagsOpen(tags_file, NULL);
+	const gchar *lang_kinds = tm_ctags_get_lang_kinds(lang);
+	GPtrArray *unknown_fields = g_ptr_array_new_full(10, NULL);
```

Comment:
any reason not to use `GArray` instead?  OK it's a tad awkward as well, but less so than `GPtrArray` for storing integers I'd say.  I don't have to draw you a picture, but:

```c
GArray *array = g_array_sized_new(FALSE, FALSE, sizeof(guint), 10);
g_array_append_val(array, i);
guint i2 = g_array_index(array, guint, idx);
```

Reply:
The fundamental reason was that I never used GArray ;-). I've updated the code with your suggestion.